engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java

```diff
@@ -118,22 +112,22 @@ public void activate(final long key, final JobRecord record) {
     validateParameters(type);
     EnsureUtil.ensureGreaterThan("deadline", deadline, 0);
 
-    resetVariablesAndUpdateJobRecord(key, record);
+    updateJobRecord(key, record);
 
     updateJobState(State.ACTIVATED);
 
     makeJobNotActivatable(type);
 
     deadlineKey.wrapLong(deadline);
-    deadlinesColumnFamily.put(deadlineJobKey, DbNil.INSTANCE);
+    deadlinesColumnFamily.insert(deadlineJobKey, DbNil.INSTANCE);
   }
 
   @Override
   public void recurAfterBackoff(final long key, final JobRecord record) {
     updateJob(key, record, State.ACTIVATABLE);
     jobKey.wrapLong(key);
     backoffKey.wrapLong(record.getRecurringTime());
-    backoffColumnFamily.delete(backoffJobKey);
+    backoffColumnFamily.deleteExisting(backoffJobKey);
   }
```

Comment:
💭 I totally agree with this change, but it made me think. What if a failed job (with backoff) is deleted before the backoff expired? I think the backoff column family entry is not deleted then. I think we have a bug there, although unrelated to this change. Might be useful to have a look or to open a bug for it. (note that this only occurs when jobs are failed with a retry which is already a niche that most users won't run into).

Reply:
@korthout Could you create an issue for this. Even with your description I am unsure how to phrase what the problem is, tbh.